### PR TITLE
implement profiler context tagging API

### PR DESCRIPTION
**Pull request by richardstartin**
# What Does This Do

Let's a user register arbitrary attributes for profiling context by setting:
`-Ddd.profiling.experimental.context.attributes=foo,bar`
Which then allows the user to call 
`ProfilingContext.get().setContextValue("foo", stringValue);`
and have the tags "foo=`stringValue`" appended to each JFR event emitted.

# Motivation

# Additional Notes

### CODE REVIEW

File: dd-trace-api/src/main/java/datadog/trace/api/config/ProfilingConfig.java [hunk 1]
```diff
@@ -159,5 +159,7 @@ public final class ProfilingConfig {
 
   public static final String PROFILING_DEBUG_DUMP_PATH = "profiling.debug.dump_path";
 
+  public static final String PROFILING_CONTEXT_ATTRIBUTES = "profiling.context.attributes";
```
Comment (jbachorik) [inline dd-trace-api/src/main/java/datadog/trace/api/config/ProfilingConfig.java:162]: Could this be an experimental key for now?
The registration process might possibly change in the future, correct?

Comment (richardstartin) [reply dd-trace-api/src/main/java/datadog/trace/api/config/ProfilingConfig.java:162]: yes, it should be experimental, thanks.

Comment (richardstartin) [reply dd-trace-api/src/main/java/datadog/trace/api/config/ProfilingConfig.java]: Changed.

**Comment by richardstartin** [comment]
I think we need to merge https://github.com/DataDog/java-profiler-library/pull/46 before this, because the API is a little awkward to use. I'd like to restore some of the modularity we previously broke for the sake of saving a thread-local lookup for the tid (not a good trade off).

**Review by bantonsson — APPROVED** [review]
Looks good from the tracer side

**Pull request merged**